packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2851,6 +2842,43 @@ export class ContainerRuntime
 		}
 	}
 
+	private scheduleFlush() {
+		if (this.flushTaskExists) {
+			return;
+		}
+
+		this.flushTaskExists = true;
+		const flush = () => {
+			this.flushTaskExists = false;
+			this.flush();
+		};
+
+		switch (this.flushMode) {
+			case FlushMode.TurnBased:
+				Promise.resolve()
+					.then(() => flush())
+					.catch((error) => {
+						this.closeFn(error as GenericError);
+					});
+				break;
+			case FlushMode.Async:
+				setTimeout(() => {
+					try {
+						flush();
+					} catch (error) {
+						this.closeFn(error as GenericError);
+					}
+				}, 0);
+				break;
+			default:
+				assert(
```

Comment:
I assume there is no change in behavior (despite code changes) as we do flush at the end of last orderSequentially, right?